src/main/java/org/qiyana/model/Product.java

```diff
@@ -0,0 +1,48 @@
+package org.qiyana.model;
+
+import javax.persistence.*;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import java.util.Currency;
+
+@Entity
+@NoArgsConstructor
+@Getter
+public class Product {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  private String name;
+  private String description;
+  private double price;
+  private Currency currency;
+
+  public Product(Long id, String name, String description, double price, String currencyCode) {
+    this.id = id;
```

Comment:
`id` and any other `GeneratedValue` should never be a constructor parameter. You will pretty much always pass `null` here.